src/abstractdataframe/selection.jl

```diff
@@ -155,7 +196,8 @@ function select_transform!(nc::Pair{<:Union{Int, AbstractVector{Int}},
         allow_resizing_newdf[] = false
         respar = parent(res)
         if copycols && !(fun isa ByRow) &&
-            (res isa SubArray || any(i -> respar === parent(cdf[i]), col_idx))
+            (res isa SubArray || any(i -> respar === parent(cdf[i]),
+                col_idx isa AsTable ? col_idx.colselector : col_idx))
```

Comment:
It would be more readable to define an intermediary variable.

Reply:
changed